src/main/java/mServer/crawler/sender/MediathekArd.java

```diff
@@ -47,572 +43,572 @@
-            }
+    private void addTage() {
+      // http://www.ardmediathek.de/tv/sendungVerpasst?tag=0 ... 6
+      for (int i = 0; i <= 6; ++i) {
+        if (Config.getStop()) {
+          break;
+        }
+        final String urlTage = "http://www.ardmediathek.de/tv/sendungVerpasst?tag=" + i;
+        final GetUrl getUrl = new GetUrl(getWartenSeiteLaden());
+        seite1 = getUrl.getUri(SENDERNAME, urlTage, StandardCharsets.UTF_8, 2, seite1, "");
+        if (seite1.length() == 0) {
+          Log.errorLog(765323214, "Leere Seite: " + urlTage);
+          return;
         }
         int pos = 0;
-        String url = "";
-        while (!Config.getStop() && (pos = seite.indexOf(MUSTER_URL_THEMA, pos)) != -1) {
+        String url, datum, zeit = "", titel, dauer, urlSendung, thema;
+        long d = 0;
+        while (!Config.getStop() && (pos = seite1.indexOf(MUSTER_ADD_TAGE, pos)) != -1) {
+          zeit = seite1.extract("<span class=\"date\">", "<", pos) + ":00";
+          pos += MUSTER_ADD_TAGE.length();
+
+          url = seite1.extract("documentId=", "&", pos);
+          if (url.contains("\"")) {
+            url = url.substring(0, url.indexOf('\"'));
+          }
+          if (!url.isEmpty()) {
+            url = url.replace("&amp;", "&");
+            thema = seite1.extract("<span class=\"titel\">", "<", pos);
+            if (thema.endsWith("Uhr") && thema.contains(",")) {
+              // tagesschau, 09:00 Uhr
+              thema = thema.substring(0, thema.indexOf(','));
+            }
+            datum = seite1
+                .extract("<title>Videos (TV-Sendungen) des Senders Das Erste vom", "- ARD").trim();
+            titel = seite1.extract("<h4 class=\"headline\">", "<", pos);
+            dauer = seite1.extract("<p class=\"subtitle\">", "<", pos);
             try {
-                pos += MUSTER_URL_THEMA.length();
-                int pos1 = pos;
-                int pos2 = seite.indexOf("\"", pos);
-                if (pos1 != -1 && pos2 != -1) {
-                    url = seite.substring(pos1, pos2);
-                }
-                if (!url.isEmpty()) {
-                    url = "http://www.ardmediathek.de/tv/sendungen-a-z?buchstabe=" + url;
-                    feedSuchen1(url);
-                }
-            } catch (Exception ex) {
-                Log.errorLog(698732167, ex, "kein Thema");
+              final Matcher dauerMatcher = Pattern.compile(DAUER_REGEX_PATERN).matcher(dauer);
+              if (dauerMatcher.find()) {
+                d = Long.parseLong(dauerMatcher.group()) * 60;
+              }
+            } catch (final Exception ignored) {
+            }
+            if (d == 0) {
+              Log.errorLog(915263621, "Dauer==0: " + urlTage);
             }
+            urlSendung = seite1.extract("<a href=\"/tv/", "\"", pos);
+            if (!urlSendung.isEmpty()) {
+              urlSendung = "http://www.ardmediathek.de/tv/" + urlSendung;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "http://www.ardmediathek.de/tv/" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)